src/providers/twitch/TwitchMessageBuilder.cpp

```diff
@@ -1021,24 +1023,44 @@ Outcome TwitchMessageBuilder::tryAppendEmote(const EmoteName &name)
 
     const auto &globalBttvEmotes = app->twitch->getBttvEmotes();
     const auto &globalFfzEmotes = app->twitch->getFfzEmotes();
+    const auto &globalSeventvEmotes = app->twitch->getSeventvEmotes();
 
     auto flags = MessageElementFlags();
     auto emote = boost::optional<EmotePtr>{};
 
     // Emote order:
     //  - FrankerFaceZ Channel
+    //  - 7TV Channel
     //  - BetterTTV Channel
+    //  - 7TV Global
     //  - FrankerFaceZ Global
     //  - BetterTTV Global
```

Comment:
This order doesn't match the concept BTTV -> FFZ -> 7TV, but that's how it was implemented in the fork.

Reply:
I propose the following emote order:
FrankerFaceZ Channel -> BetterTTV Channel -> 7TV Channel
FrankerFaceZ Global -> BetterTTV Global -> 7TV Global

The global order change will not make any difference for users, the channel order *should* not make any difference, unless a user has two emotes added with the same name.

Reply:
With what can be seen so far during this halloween season, it seems many streamers are taking advantage of emote sets and the fact 7TV overrides the other providers.

Perhaps worth re-thinking the proposed order for channel emotes?

Reply:
> Perhaps worth re-thinking the proposed order for channel emotes?

Why don't they just.... you know.... not use emotes with the same name?

If they find it annoying to swap emotes out during seasonal events, then they should use 7TV as it has the best features for emote-set-swapping, but I don't see the argument for priority when it comes to something the user chooses to do.

Global emotes on the other hand would be a better argument.

Reply:
> Why don't they just.... you know.... not use emotes with the same name?

For people that upload emotes it sucks, I would have to delete emotes with a lot of channels using it from BTTV, since they have limited upload slots, and it auto-enables on the channel

Reply:
> For people that upload emotes it sucks, I would have to delete emotes with a lot of channels using it from BTTV, since they have limited upload slots, and it auto-enables on the channel

Is this issue resolved by 7TV? So the same scenario but backwards won't happen to 7TV emote uploaders?

Reply:
> Is this issue resolved by 7TV? So the same scenario but backwards won't happen to 7TV emote uploaders?

Yeah, it actually is, you don’t have to enable your own emotes in 7tv, you can just disable them but keep them uploaded.

Reply:
> Is this issue resolved by 7TV? So the same scenario but backwards won't happen to 7TV emote uploaders?

It won't happen, since I can create and upload emotes, and they don't take slots in my channel, BTTV after the change merged emotes created and added to the channel, it's really stupid